src/app/ReadClient.cpp

```diff
@@ -187,6 +188,12 @@ void ReadClient::Close(CHIP_ERROR aError, bool allowResubscription)
             if (allowResubscription &&
                 (mReadPrepareParams.mEventPathParamsListSize != 0 || mReadPrepareParams.mAttributePathParamsListSize != 0))
             {
+                if (aError == CHIP_ERROR_LIT_SUBSCRIBE_INACTIVE_TIMEOUT)
```

Comment:
Again: what if our client wants us to re-subscribe immediately because it knows the LIT is active?  We are setting the state here in a way that cannot be controlled by the client, right?  If the client then does "ok, go ahead and re-subscribe now" what happens?

Again: it seems to me that we should set this state _when we are actually in this state_.  That is, when we know we skipped the resubscribe but want to trigger it on activity.

If we need a way for the client to put us into that state, them maybe what we should have is that OnResubscriptionNeeded can return CHIP_ERROR_LIT_SUBSCRIBE_INACTIVE_TIMEOUT when it's skipping things, and then this caller would enter this state when that error code is returned?  This would obviously need to get documented in ReadClient::Callback. Then in DefaultResubscribePolicy we return this error code.  Needs checking to see whether it all actually works, of course; I did not carefully audit the code path for the resubscribe notification just now.

But the key part is this: the API needs to both allow the API consumer to put us in the InactiveICDSubscription state and allow them to _not_ go into that state and just re-subscribe.... 

Another option is to got into that state here but have a new return value here the API consumer can return that means "actually, go ahead now".

Of course if there is some way the API consumer can already accomplish this "just try to resubscribe now", that's great, but that way needs to be clearly documented.

Reply:
Makes sense. I moved the this line after `OnResubscriptionNeeded`.